exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ColumnDefault.kt

```diff
@@ -0,0 +1,159 @@
+package org.jetbrains.exposed.sql
+
+/**
+ * Represents a default value for a column.
+ *
+ * Commonly it should be implementation of [ClientColumnDefault] or [DatabaseColumnDefault] interfaces.
+ */
+interface ColumnDefault<T>
+
+/**
+ * Represents a client-side default value for a column.
+ *
+ * The column with [ClientColumnDefault] default value will not affect DDL,
+ * and will be automatically added to all the insert/upsert statements
+ */
+interface ClientColumnDefault<T> : ColumnDefault<T>
+
+/**
+ * Represents a database-side default value for a column.
+ *
+ * The [DatabaseColumnDefault] default value will be used for DDL generation,
+ * and will not be added to insert/upsert statements automatically.
+ */
+interface DatabaseColumnDefault<T> : ColumnDefault<T>
+
+/**
+ * Represents a default value for a column generated by an SQL expression.
+ */
+interface ColumnDefaultExpression<T> : ColumnDefault<T> {
+    val expression: Expression<T>
+}
+
+/**
+ * Represents a default value for a column generated by a function.
+ */
+interface ColumnDefaultValue<T> : ColumnDefault<T> {
+    val value: () -> T
```

Comment:
I would make a method instead cause functional fields are harder to modify without breaking change